src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -73,6 +73,11 @@ public override BoundNode VisitDiscardPattern(BoundDiscardPattern node)
             return null;
         }
 
+        public override BoundNode VisitSlicePattern(BoundSlicePattern node)
+        {
+            return null;
```

Comment:
Why are we not visiting the nested patterns?

Reply:
They indeed need to be visited. This will be addressed in a later PR (here's the relevant commit I'm working on): https://github.com/dotnet/roslyn/commit/933dfd88a0dc86095d886979e16a2734b6d41da1
With the current PR, multiple nullability scenarios will hit assertions and/or produce incorrect nullability warnings.